src/main/java/spoon/support/visitor/java/JavaReflectionTreeBuilder.java

```diff
@@ -141,6 +136,24 @@ public <T, R extends CtType<T>> R scan(Class<T> clazz) {
 		}
 	}
 
+	private <T> CtPackage getCtPackage(Class<T> clazz) {
+		Package javaPackage = clazz.getPackage();
+		if (javaPackage == null && clazz.isArray()) {
+			javaPackage = getArrayType(clazz).getPackage();
+		}
+		if (javaPackage == null) {
+			return factory.Package().getRootPackage();
+		}
+		return factory.Package().getOrCreate(javaPackage.getName());
+	}
+
+	private static Class<?> getArrayType(Class<?> array) {
```

Comment:
The naming here is confusing, maybe something like `getInnermostComponentType` would be better?

Reply:
The rest of the code `CtArrayReference` uses the same terminology, so I opted to keep it the same. But I can adjust it, sure :)

Reply:
I see... well, then it's fine I guess.